draft-ietf-rtcweb-mdns-ice-candidates.md

```diff
@@ -345,8 +366,45 @@ IP addresses that this specification is designed to hide. The use of
 registered mDNS hostnames SHOULD be scoped by origin, and SHOULD have the
 lifetime of the page.
 
+Determination of Address Privacy
+--------------------------------
+
+Naturally, an address that is already exposed to the Internet does not need to
+be protected by mDNS, as it can be trivially observed by the web server or
+remote endpoint. However, determining this ahead of time is not straightforward;
+while the fact that an IPv4 address is private can sometimes be inferred by its
+value, e.g., whether it is an {{RFC1918}} address, the reverse is not
+necessarily true. IPv6 addresses present their own complications, e.g.,
+private IPv6 addresses as a result of NAT64 {{RFC6146}}.
+
+Instead, the determination of whether an address is public can be reliably made
+as part of the ICE gathering process, namely, if the query to the
+STUN {{RFC5389}} server returns the same value as the local address. This can
+be done for both IPv4 and IPv6 local addresses, provided that the application
+has configured both IPv4 and IPv6 STUN servers. If this situation occurs, i.e.,
+STUN returns the same IP address value for an address that has already
+been communicated as an mDNS candidate during the current ICE gathering phase,
+the ICE agent MUST NOT eliminate the candidate as redundant and MUST send
+the IP address as a server-reflexive candidate. This allows the ICE agent
+to send mDNS candidates immediately (i.e., without waiting for STUN),
+even if the associated addresses may not be private.
+
+Once an address has been identified as public, the ICE agent MAY cache this
+information and omit mDNS protection for that address in future ICE gathering
+phases.
+
+Special Handling for IPv6 Addresses
+-----------------------------------
+
+As noted in {{IPHandling}}, private IPv4 addresses are especially problematic
+because of their unbounded lifetime. However, the {{RFC4941}} IPv6
+addresses recommended for WebRTC have inherent privacy protections, namely
+a short lifetime and the lack of any stateful information. Accordingly,
+implementations MAY choose to not obscure {{RFC4941}} addresses with mDNS names
```

Comment:
We use 'obscure', 'conceal' and maybe 'obfuscate' as well.
We should stick with one term I guess.
Can be left to a follow-up editorial PR.

Reply:
Switched to *conceal*, see also #77.